mantidproject/mantid
# Usage tracking of algorithms

Implementing the approved design https://github.com/mantidproject/documents/blob/master/Design/Usage/AlgorithmUsageStatistics.md

Add tracking of algorithm usage as well as rationalising the start-up tracking and adding functionality.

Algorithms are (if usagereporting is enabled) counted and updated either 1/minute, or on shutdown to the database.

The back end is visible here:
http://reports.mantidproject.org/api/feature

Let me know when testing is complete and I will clear the tracking table to remove data from testing.

closes #12041
### Release Notes:

http://www.mantidproject.org/ReleaseNotes_3_6_Framework_Changes#Algorithms
### To Test:
1. either use a production version of Mantid, or set usagereports.enabled = 1 in mantid.user.properties
2. Startup mantid and run a few algorithms
3. check the counts in 
   http://reports.mantidproject.org/api/feature
   increase you may have a 1 minute delay
4. run an algorithm and then close mantidplot, check this is counted as well.
5. make sure you test previously untested algorithms as well as already existing ones.
6. set usagereports.enabled = 0 and startup mantid.  check nothing is recorded.
7. UsageService is also exposed to python, have a play with that as well.
